src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs

```diff
@@ -381,12 +382,14 @@ public static IResourceBuilder<T> WithDockerfile<T>(this IResourceBuilder<T> bui
 
         var fullyQualifiedDockerfilePath = Path.GetFullPath(dockerfilePath, fullyQualifiedContextPath);
 
-        var imageName = builder.GenerateImageName();
+        var imageName = ImageNameGenerator.GenerateImageName(builder);
```

Comment:
Do we want to avoid overwriting the image name?

Reply:
Oh you mean if someone called `WithImage` before calling this?